util/network/cniprovider/createns_linux.go

```diff
@@ -38,63 +38,51 @@ func cleanOldNamespaces(c *cniProvider) {
 	}()
 }
 
-func createNetNS(c *cniProvider, id string) (string, error) {
+// unshareAndMount needs to be called in a separate thread
+func unshareAndMountNetNS(target string) error {
+	if err := syscall.Unshare(syscall.CLONE_NEWNET); err != nil {
+		return err
+	}
+
+	return syscall.Mount("/proc/thread-self/ns/net", target, "", syscall.MS_BIND, "")
```

Comment:
`/proc/thread-self` is only available from Linux 3.17. `fmt.Sprintf("/proc/self/task/%d/ns/net", runtime.Gettid())` costs an extra syscall, but is more widely compatible.

Reply:
Updated to a different path. We make no guarantees that buildkit works on 3.17, though.